### PR TITLE
Updating build and configuration process (#31)

### PR DESCRIPTION
**Pull request by ssciolla**
This PR includes updates and modifications to the build and configuration process for the project in order to bring it in line with other T&L applications and enable integration with the Jenkins job process. Other changes are included to fix a bug related to sending emails and make formatting more consistent. Additional details will be provided below shortly. This PR aims to resolve issue #31 and work towards the resolution of #30.

* Previously, the `configuration/` directory contained a number of somewhat redundant versions of `config.py`, which lays out variables as part of `Application`, `Canvas`, and `ArcGIS` classes. When deployed, some of these values were overwritten using a `secrets.py` file provided as a OpenShift secret. I consolidated the different `config.py` versions into one file, and I made configurable values that might vary between them or that are sensitive through an `env.json` file loaded at the top of `config.py`. Again, this arrangement is more similar to other T&L applications' configuration process and reduces the number of files included in the project. Different versions of `env.json` can/will be stored elsewhere for developers' use.

* The previous build process used Python 3.5, leveraged Anaconda to access the ArcGIS API, and set up `cron` for use within the project. As a result of T&L's current use of Jenkins for automating jobs and the ability to now install `arcgis` through `pip`, I was able to significantly simplify the `Dockerfile` and delete several `.sh` scripts and other artifacts related to `cron`. Note that I elected to install `arcgis` using the `--no-deps` flag (see `Dockerfile` and note in `requirements.txt`), as we are only using core functionality of the module; this will hopefully result in shorter build times and smaller images. While I tried including `arcgis` with the flag within `requirements.txt`, I ran up against build errors, which led me to installing it separately. I also replaced the existing `startup.sh` script with a very simple `start.sh` script that simply determines whether to run `main.py` with or without the `--email` flag. We can create separate `.yaml` files (that set the `SEND_EMAIL` variable) and Jenkins jobs to replicate the previous `cron` schedules, if that is desired.

* I made two significant changes in `main.py`.
	* First, I changed the `getCourseIDsFromConfigCoursePage` function so that name of the course configuration page was truly configurable (`course-ids` was hard-coded before, despite there being a variable for it in `config.py`) and that the base URL for courses listed on that page could also be set (the program assumed `umich.instructure.com` before). These modifications required that I add a new parameter to the function for the `config_course_page_name`.
	* Second, when testing the email sending functionality, I noted that the Canvas object for users no longer had a `sis_login_id` attribute but did have a `login_id`. I'm not sure when or why this change occurred, but I updated the usage accordingly. 

Note: Throughout the project, I made some formatting changes to `.py` files to streamline module imports and make line spacing conformant with PEP. I've noted in issue #33 that naming conventions do not use PEP, but fixing that would require a significant re-write and may not be desired by other T&L developers.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,42 +1,27 @@
...
-##### Improved code tracebacks for exceptions
-import traceback
 
+# global variables and logging setup
 def handleError(self, record):  # @UnusedVariable
     traceback.print_stack()
+
+
+logger = logging.getLogger(__name__)
+loggingLevel = None
```
Comment (zqian) [inline main.py]: Why the level is set to None here? Log message should help.

File: main.py [hunk 2]
```diff
@@ -331,18 +318,21 @@ def closeAllCourseLogHandlers():
...
 
-def getCourseIDsFromConfigCoursePage(canvas, courseID):
+def getCourseIDsFromConfigCoursePage(canvas, courseID, config_course_page_name):
     """Read hand edited list of Canvas course ids to process from a specific Canvas course page."""
-    
-    VALID_COURSE_URL_REGEX = '^https://umich\.instructure\.com/courses/[0-9]+$'
-    pages = canvas.getCoursesPagesByNameObjects(courseID, 'course-ids')  # type: list of CanvasObject
-    courseIDs = None
 
+    regex_base = config.Canvas.BASE_URL.replace(".", "\\.")
```
Comment (zqian) [inline main.py]: good to use the base url to prepare for different deploy environment!

File: main.py (hunk 1 above)
Comment (ssciolla) [reply main.py]: `loggingLevel` is just initialized as `None`; it is set in the lines that follow. I'm not absolutely sure why it's done this way, but I'll hazard a guess it has something to do with some of the other logging settings used.

Comment (zqian) [reply main.py]: OK. I think the following code in the file is redundant.

if loggingLevel is None:
    loggingLevel = config.Application.Logging.DEFAULT_LOG_LEVEL

I think we can just use config.Application.Logging.DEFAULT_LOG_LEVEL

Comment (ssciolla) [reply main.py]: K, that makes sense. See [latest commit](https://github.com/tl-its-umich-edu/kartograafr/pull/36/commits/c8126ef4fdd6f0ed9b360a9edb0492001c1a47b6).

**Comment by ssciolla** [comment]
@zqian instructed me to merge so we can begin testing in the AWS OpenShift environment.

**Pull request merged**